### PR TITLE
feat: add WnSeparator component

### LINKED ISSUE #68: Separator
https://www.figma.com/design/J9pCZpUhcm0MRs7dFX7LTN/01.-Application-Components?node-id=146-2824&t=KQVsPbazOVcgDTKf-0

### PR DESCRIPTION
**Pull request by erskingardner**
## Summary

- Implements the Separator design component from Figma (issue #68)
- Adds `WnSeparator` widget with horizontal and vertical orientation support
- Uses `borderTertiary` color from theme by default (matches Figma spec)

## Features

- `WnSeparatorOrientation.horizontal` (default) - 1px horizontal line
- `WnSeparatorOrientation.vertical` - 1px vertical line
- Configurable `indent`, `endIndent`, and `color`

## Testing

- 33 comprehensive widget tests covering all orientations, properties, and edge cases
- All precommit checks pass

Closes #68

## Summary by CodeRabbit

* **New Features**
  * Added a customizable separator widget supporting horizontal and vertical orientations, responsive sizing, adjustable indentation, thickness, and color.

* **Tests**
  * Added comprehensive tests covering rendering, sizing, indentation, color variants, edge cases, and layout behavior in different containers.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: lib/widgets/wn_separator.dart [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      height: 1.r,
+      margin: EdgeInsets.only(left: indent.w, right: endIndent.w),
+      color: separatorColor,
+    );
+  }
+
+  Widget _buildVertical(Color separatorColor) {
+    return Container(
+      width: 1.r,
+      margin: EdgeInsets.only(top: indent.w, bottom: endIndent.w),
```
Comment (untreu2) [inline lib/widgets/wn_separator.dart]: This should be indent.h, bottom: endIndent.h

**Review by untreu2 — CHANGES_REQUESTED** [review]
